python/searching/StringBinarySearch.py

```diff
@@ -0,0 +1,55 @@
+# function to decide the flow of search
```

Comment:
Can you add a comment block explaining what is String Binary Search, along with some test cases, which would then be validated by this code block